Update search command docs (#12271)
pypa/pip: docs/html/cli/pip_search.rst

==========
pip search
==========

> .. deprecated:: 20.3.4

Reviewer: This is not deprecated in pip, at this time, so it'd be preferable to remove this line.

```suggestion
```

Description

> ===========
> .. warning::

Reviewer: ```suggestion
===========

.. attention::
```

Let's give the heading some space, and change this from a warning to something more appropriate. :)

    PyPI no longer supports ``pip search`` (or XML-RPC search). Please use https://pypi.org/search (via a browser)
    instead. See https://warehouse.pypa.io/api-reference/xml-rpc.html#deprecated-methods for more information.

>     This command may still be supported by indexes other than PyPi.

Reviewer: ```suggestion

    However, XML-RPC search (and this command) may still be supported by indexes other than PyPI.
```

Let's make this a separate sentence and explicitly call out XML-RPC search. Also, we should fix the capitalisation for PyPI here. :)

==========
pip search
==========
.. deprecated:: 20.3.4

Description
===========

> PyPI no longer supports 'pip search' (or XML-RPC search). Please use https://pypi.org/search (via a browser)

Reviewer 2: ```suggestion
PyPI no longer supports ``pip search`` (or XML-RPC search). Please use https://pypi.org/search (via a browser)
```
